huaweicloud/config/config.go

```diff
@@ -437,7 +438,7 @@ func (c *Config) NewServiceClient(srv, region string) (*golangsdk.ServiceClient,
 }
 
 func (c *Config) newServiceClientByName(client *golangsdk.ProviderClient, catalog ServiceCatalog, region string) (*golangsdk.ServiceClient, error) {
-	if catalog.Name == "" || catalog.Version == "" {
+	if catalog.Name == "" && catalog.Version == "" {
```

Comment:
as catalog.Name is required, suggest to write like this:
if catalog.Name == "" {
    ...
}